src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -971,5 +974,18 @@ private IEnumerable<ConfigurationSetting> GetCurrentKeyValueCollection(string ke
 
             return currentKeyValues;
         }
+
+        private void EnsureTracing()
+        {
+            if (!_isTracingSet)
+            {
+                // FeatureManagement assemblies may not be loaded on provider startup, so version information is gathered upon first refresh for telemetry
+                _requestTracingOptions.FeatureManagementVersion = TracingUtils.GetAssemblyVersion(RequestTracingConstants.FeatureManagementAssemblyName);
+
+                _requestTracingOptions.FeatureManagementAspNetCoreVersion = TracingUtils.GetAssemblyVersion(RequestTracingConstants.FeatureManagementAspNetCoreAssemblyName);
+
+                _isTracingSet = true;
```

Comment:
Do you mind setting this as the first step? Usually a good idea for once only behavior.